spec/requests/api/v1/docs/pages_spec.rb

```diff
@@ -34,6 +35,69 @@
         end
       end
     end
+
+    describe "create a new page" do
+      post("pages") do
+        tags "pages"
+        description(<<-DESCRIBE.strip)
+        This endpoint allows the client to create a new page.
+        DESCRIBE
+
+        produces "application/json"
+        consumes "application/json"
+
+        parameter name: :page, in: :body, schema: {
+          type: :object,
+          properties: {
+            title: { type: :string, description: "Title of the page" },
+            slug: { type: :string, description: "Used to link to this page in URLs, must be unique and URL-safe" },
+            description: { type: :string, description: "For internal use, helps similar pages from one another" },
+            body_markdown: { type: :string, description: "The text (in markdown) of the ad (required)" },
+            body_json: { type: :string, description: "For JSON pages, the JSON body" },
+            is_top_level_path: { type: :boolean,
+                                 description: "If true, the page is available at '/{slug}' instead of '/page/{slug}', use with caution" },
+            template: {
+              type: :string, enum: Page::TEMPLATE_OPTIONS, default: "contained",
+              description: "Controls what kind of layout the page is rendered in"
+            },
+
+            required: %w[title slug description template]
+          }
```

Comment:
The `required` keyword should live outside the `properties` hash within the schema definition because otherwise we get this in the swagger editor

<img width="1323" alt="Screen Shot 2023-01-26 at 10 49 16" src="https://user-images.githubusercontent.com/6045239/214897308-6568ca54-0977-4807-ab59-8470ae6eeb07.png">

```suggestion
            }
          },
          required: %w[title slug description template]
```

Reply:
Great catch, updated!